go/worker/compute/executor/committee/node.go

```diff
@@ -432,11 +436,21 @@ func (n *Node) HandleEpochTransitionLocked(epoch *committee.EpochSnapshot) {
 	if !epoch.IsExecutorWorker() {
 		// Clear incoming queue if we are not a worker anymore.
 		n.scheduler.Clear()
+		n.lastScheduledCache.Clear()
 		incomingQueueSize.With(n.getMetricLabels()).Set(0)
 	}
 
 	switch {
 	case epoch.IsExecutorMember():
+		if state, ok := n.state.(StateWaitingForFinalize); ok {
```

Comment:
Hm on a related note, in `HandleNewBlockLocked` when in `StateWaitingForFinalize` state we currently always call `removeTxBatch` but we likely need to check that the round succeeded (header type is "normal") and that this was actually the round that we processed (e.g. by comparing the IO roots) as otherwise we can incorrectly remove a batch that was not actually finalized?

Reply:
Yeah good point, i did notice that, but then only checked what happens on epoch transition, which is handled fine (we don't delete the batch since we transition the state before that already), didnt thought about other failures though. Will fix it

Reply:
done

Reply:
Looks good, thanks!